### PR TITLE
fix: use search subdir imports and split subtools

### PR DESCRIPTION
**Pull request by njhale**
- Use the new subdirectory based import paths for `search` tools
- Split the tools in this repository into separate subdirectories to follow new best practices

Note:

Since the `html2text` tool is now referenced by other tools in this repo using its GitHub import path, the UX around testing local changes to it becomes kludgy; i.e. you need to change the reference to a local file path, then change it back to the GitHub reference before merging any changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,26 +23,26 @@ There has been speculation about whether George Mallory and Andrew Irvine might
...
 
 ```yaml
-tools: question-answerer-ddg from github.com/gptscript-ai/question-answerer
+tools: github.com/gptscript-ai/question-answerer/duckduckgo
 
 Who were the first people to climb Mount Everest? Use the question-answerer tool to answer.
 ```
 
-You can also clone the repo and run the tool:
+You can also clone the repo and run the `html2text` tool:
```
Comment (njhale) [inline README.md:38]: not sure how useful this is

Comment (g-linville) [reply README.md:38]: yeah we can just remove this part

**Review by g-linville — APPROVED** [review]
thanks @njhale

File: README.md [hunk 2]
```diff
@@ -23,26 +23,26 @@ There has been speculation about whether George Mallory and Andrew Irvine might
 
 ## Usage
 
-The easiest way to use these tools is to reference them by their GitHub repo in a command or in a script:
+The easiest way to use these tools in a command or `.gpt` script is to reference them by path their GitHub repo and path in a command or in a script:
```
[the remark this answers is not in the document]
Comment (njhale) [reply README.md]: 👍

File: README.md [hunk 3]
```diff
@@ -23,34 +23,14 @@ There has been speculation about whether George Mallory and Andrew Irvine might
 
 ## Usage
 
-The easiest way to use these tools is to reference them by their GitHub repo in a command or in a script:
+The easiest way to use these tools in a command or `.gpt` script is to reference them by path their GitHub repo and path:
```
Comment (g-linville) [inline README.md:26]:
```suggestion
The easiest way to use these tools in a command or `.gpt` script is to reference them by their GitHub repo and path:
```

**Review by g-linville — CHANGES_REQUESTED** [review]

**Review by g-linville — APPROVED** [review]

**Pull request merged**